src/node_config.rs

```diff
@@ -283,6 +297,37 @@ fn parse_config(config_path: &str) -> NodeConfig {
     config
 }
 
+fn run_derive_pk(matches: &ArgMatches) -> ! {
+    let sk_passed_as_arg = matches.occurrences_of("secret-key") > 0;
+    if sk_passed_as_arg {
+        warn!("You should not pass the secret key via arguments due to \
+               security reasons. Use the environment variable instead");
+    }
+
+    let pk_from_arg = matches.value_of("secret-key").map(|s| {
+        let sk_bytes: [u8; 32] = FromHex::from_hex(s).expect("Invalid DHT secret key");
+        let sk = SecretKey::from_slice(&sk_bytes).expect("Invalid DHT secret key");
+        sk.public_key()
+    });
+    let pk_from_file = matches.value_of("keys-file").map(|keys_file| {
+        let mut file = std::fs::File::open(keys_file).expect("Failed to read the keys file");
```

Comment:
Should I move it to a standalone function or is ^C^V fine?

Reply:
It won't harm if you can :)
Maybe also move it to main.rs to solve the problem with log?

Reply:
Let's do this in a separate PR